spyder/widgets/tests/test_status.py

```diff
@@ -62,53 +66,75 @@ def test_status_bar_widget_signal(status_bar, qtbot):
 
     win, statusbar = status_bar
-    w = CondaStatus(win, statusbar)
+    w = InterpreterStatus(win, statusbar)
 
     # Check that we process stdout and stderr correctly
     out_or_err = mock_py_ver + ' (hello!)'
     for (out, err) in [(out_or_err, ''), ('', out_or_err)]:
         # We patch the method that calls for info to return values to test
-        mocker.patch.object(w, '_get_conda_env_info', return_value=(out, err))
+        mocker.patch.object(w, '_get_interpreter_env_info',
+                            return_value=(out, err))
 
         if os.name == 'nt':
             interpreter = os.sep.join(['miniconda', 'python'])
         else:
             interpreter = os.sep.join(['miniconda', 'bin', 'python'])
         w.update_interpreter(interpreter)
+        expected = 'conda: base (Python 6.6.6)'
         assert w.get_tooltip() == interpreter
-        text = w._process_conda_env_info()
-        assert 'base' in text
-        assert mock_py_ver in text
+        assert expected == w._process_interpreter_env_info()
 
         # We patch the method that calls for info to return values to test
         if os.name == 'nt':
             interpreter = os.sep.join(['miniconda', 'envs', 'foo', 'python'])
         else:
-            interpreter = os.sep.join(['miniconda', 'envs', 'foo', 'bin', 'python'])
+            interpreter = os.sep.join(['miniconda', 'envs', 'foo', 'bin',
+                                       'python'])
+
         w.update_interpreter(interpreter)
+        expected = 'conda: foo (Python 6.6.6)'
         assert w.get_tooltip() == interpreter
-        text = w._process_conda_env_info()
-        assert 'foo' in text
-        assert mock_py_ver in text
+        assert expected == w._process_interpreter_env_info()
 
 
-def test_status_bar_no_conda_status(status_bar, qtbot, mocker):
-    mocker.patch.object(utils, 'is_anaconda', return_value=False)
+def test_status_bar_other_interpreter_status(status_bar, qtbot, mocker):
+    mocker.patch.object(spyder.widgets.status, 'is_conda_env',
+                        return_value=False)
 
     win, statusbar = status_bar
-    w = CondaStatus(win, statusbar)
-    mocker.patch.object(w, '_get_conda_env_info',
+    w = InterpreterStatus(win, statusbar)
+    mocker.patch.object(w, '_get_interpreter_env_info',
                         return_value=('Python 3.6.6', ''))
 
     interpreter = os.sep.join(['some-other', 'bin', 'python'])
     w.update_interpreter(interpreter)
     assert w.get_tooltip() == interpreter
-    assert '' == w._process_conda_env_info()
+    assert 'venv (Python 3.6.6)' == w._process_interpreter_env_info()
+
+
+@pytest.mark.skipif(sys.platform != 'darwin', reason="Only valid on Mac")
+def test_status_bar_internal_interpreter_status(status_bar, qtbot, mocker):
+    mocker.patch.object(spyder.widgets.status, 'is_conda_env',
+                        return_value=False)
+    mocker.patch.object(spyder.widgets.status, 'running_in_mac_app',
```

Comment:
It should also be possible to patch `running_in_mac_app` by importing it from the module it is defined.